src/threadpool.c

```diff
@@ -261,6 +268,62 @@ static void init_once(void) {
   init_threads();
 }
 
+int uv_set_threadpool_size(unsigned int number) {
+  unsigned int spin_nthreads;
+
+  /* Ensure the threadpool is initialized */
+  if (nthreads == 0)
+    uv_once(&once, init_once);
+
+  /* no-op if the threadpool size is already set */
+  if (number == nthreads)
+    return 0;
+
+
+  if (nthreads < number) {
+    /* Just spin up the difference and respect the MAX_THREADPOOL_SIZE */
+    spin_nthreads = number - nthreads;
+    if (spin_nthreads > MAX_THREADPOOL_SIZE)
+      spin_nthreads = MAX_THREADPOOL_SIZE - nthreads;
+  } else {
+    return UV_EINVAL; /* Cannot shrink the threadpool */
+  }
+
+  /* Alloc -with the expected size- if was not alloc(ed) before */
+  if (threads == default_threads) {
```

Comment:
I need to check but I think this will need a lock around it...

Reply:
Is it possible that the tear down call needs a mutex as well?

Reply:
This operation needs it, not the teardown since it happens at library destruction time.